operators/tokenizer/bpe_utils.hpp

```diff
@@ -359,6 +359,112 @@ class PreTokenizerWithRegEx {
     return {};
   }
 
+  void CategoryMatch(size_t& index, std::set<ufal::unilib::unicode::category_t>& categories){
+    while (categories.find(ufal::unilib::unicode::category(m_text[index])) != categories.end()){
+      index++;
```

Comment:
Need check if index is greater than m_text.size()

Reply:
right, good catch